### PR TITLE
Improve convertToDocx task

### PR DESCRIPTION
**Pull request by michael-simons**
* Use doFirst to make sure directory is created
* Refer to generateDocBook for the output dir to avoid path duplication

### CODE REVIEW

**Comment by gernotstarke** [comment]
danke Michael!

File: folge-3/build.gradle [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import org.asciidoctor.gradle.AsciidoctorTask
+import java.nio.file.Paths
```
Comment (rdmueller) [inline folge-3/build.gradle:2]: Why do you need an additional import?

File: folge-3/build.gradle [hunk 2]
```diff
@@ -69,9 +70,11 @@ task convertToDocx (
         dependsOn: 'generateDocBook',
         type: Exec
 ) {
-  workingDir 'build/asciidoc/docbook'
+  doFirst {
+    file(Paths.get(generateDocBook.outputDir.toString(), "docx")).mkdirs()
+  }
+  workingDir file(Paths.get(generateDocBook.outputDir.toString(), "docbook"))
```
Comment (rdmueller) [inline folge-3/build.gradle:76]: ok. using `generateDocBook.outputDir` is cleaner and a `doFirst` is also cleaner but
`new File(generateDocBook.outputDir.canonicalPath+"/docx").mkdirs()`
is shorter and you can omit the include above :-)

**Review by rdmueller — COMMENTED** [review]
Thanx for your input! I will also refactor the `doFirst` in docToolchain!

File: folge-3/build.gradle (hunk 2 above)
Comment (michael-simons) [reply folge-3/build.gradle:76]: Yeah, good point.
The `doFirst` was required, though. Didn't work reliable for me in the `simple-meetup` project without.

Comment (michael-simons) [reply folge-3/build.gradle:76]: I like the suggested change. But let us agree on `new File(generateDocBook.outputDir.canonicalPath, "docx").mkdirs()` subtle, but much nicer :)

Comment (rdmueller) [reply folge-3/build.gradle:76]: that's true and I still don't understand why it does not work reliable. But with your refactoring, I guess I can fix this for docToolchain in general...

Comment (rdmueller) [reply folge-3/build.gradle:76]: we should more often do pair coding - I have the feeling that the result could be something big :-)

Comment (michael-simons) [reply folge-3/build.gradle:76]: Thanks Ralf :) much appreciated.

**Pull request merged**